Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -538,6 +542,57 @@ void MemoryWidget::OnSetValue()
   Update();
 }
 
+void MemoryWidget::OnSetValueFromFile()
+{
+  if (!Core::IsRunning())
+    return;
+
+  bool good_address, good_offset;
+  // Returns 0 if conversion fails
+  u32 addr = m_search_address->text().toUInt(&good_address, 16);
+  addr += m_search_offset->text().toUInt(&good_offset, 16);
+
+  if (!good_address)
+  {
+    ModalMessageBox::critical(this, tr("Error"), tr("Bad address provided."));
+    return;
+  }
+
+  if (!good_offset)
+  {
+    ModalMessageBox::critical(this, tr("Error"), tr("Bad offset provided."));
+    return;
+  }
+
+  QString path = QFileDialog::getOpenFileName(this, tr("Select a file"), QDir::currentPath(),
+                                              QStringLiteral("All files (*.*)"));
+  if (path.isNull())
+  {
+    ModalMessageBox::critical(this, tr("Error"), tr("No file provided."));
+    return;
+  }
+
+  File::IOFile f(path.toStdString(), "rb");
+
+  if (!f)
+  {
+    ModalMessageBox::critical(nullptr, QObject::tr("Error"), QObject::tr("Unable to open file."));
+    return;
+  }
+
+  const u64 file_length = f.GetSize();
+  std::vector<u8> file_contents(file_length);
+  f.ReadBytes(file_contents.data(), file_length);
+  const QByteArray bytes(reinterpret_cast<const char*>(file_contents.data()), file_length);
```

Comment:
I don't think using a `QByteArray` is necessary as you can loop over `std::vector<u8> file_contents` later on like this:
```c++
  for (u8 b : file_contents)
    accessors->WriteU8(addr++, b);
```